### ISSUE TITLE
Offset not working in API

### ISSUE THREAD

**Issue by nonamethanks** [Jackett/Jackett#13681]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Latest linuxserver/jackett docker instance (Jackett Version v0.20.2163).
```

### Description

I was wondering why Sonarr couldn't find the latest One Piece episode from Jackett's AnimeBytes index, and I realized that the offset was not working and all pages were returning the same result. Trying with Nyaa I get the exact same issue. Here's a simplified example:

/api/v2.0/indexers/nyaasi/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=0&limit=1&q=One%20Piece
/api/v2.0/indexers/nyaasi/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=1&limit=1&q=One%20Piece
/api/v2.0/indexers/nyaasi/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=2&limit=1&q=One%20Piece
/api/v2.0/indexers/nyaasi/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=3&limit=1&q=One%20Piece

These return the exact same torrent.  The results stay the same no matter the offset. If I set limit=100 I get the same 100 torrents no matter the offset. Via UI I'm able to find hundreds of different torrents instead. How come? 

### Logged Error Messages

No errors in the logs.

**Comment by ilike2burnthing** [comment]
As far as I'm aware `Offset` is being used for pagination in Jackett. Only 4 indexers support pagination, and only 2 of those use `query.Offset` to do so (AnimeBytes and Nyaa aren't them).

With that said, Sonarr wouldn't use it even if it were available, it wants a maximum of 100 results per indexer, and just makes one request per keyword.

**Comment by nonamethanks** [comment]
> With that said, Sonarr wouldn't use it even if it were available, it wants a maximum of 100 results per indexer, and just makes one request per keyword.

That doesn't sound right, if I look at the Sonarr logs it tried to fetch 1000 results. 

```
2022-10-25 17:32:31.4|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=0&limit=100&q=One%20Piece+1037
2022-10-25 17:32:33.6|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=100&limit=100&q=One%20Piece+1037
2022-10-25 17:32:33.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=200&limit=100&q=One%20Piece+1037
2022-10-25 17:32:35.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=300&limit=100&q=One%20Piece+1037
2022-10-25 17:32:37.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=400&limit=100&q=One%20Piece+1037
2022-10-25 17:32:38.0|Debug|RTorrentProxy|Executing remote method: d.multicall2
2022-10-25 17:32:38.1|Debug|RTorrent|Retrieved metadata of 693 torrents in client
2022-10-25 17:32:38.2|Debug|Api|[GET] /api/v3/queue/details?seriesId=53: 200.OK (1 ms)
2022-10-25 17:32:38.3|Debug|Api|[GET] /api/v3/command: 200.OK (0 ms)
2022-10-25 17:32:38.4|Debug|Api|[GET] /api/v3/queue/details?seriesId=53: 200.OK (1 ms)
2022-10-25 17:32:38.4|Debug|Api|[GET] /api/v3/command: 200.OK (0 ms)
2022-10-25 17:32:39.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=500&limit=100&q=One%20Piece+1037
2022-10-25 17:32:41.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=600&limit=100&q=One%20Piece+1037
2022-10-25 17:32:43.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=700&limit=100&q=One%20Piece+1037
2022-10-25 17:32:45.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=800&limit=100&q=One%20Piece+1037
2022-10-25 17:32:47.7|Debug|Torznab|Downloading Feed http://jackett:9117/api/v2.0/indexers/animebytes/results/torznab/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=900&limit=100&q=One%20Piece+1037
```

In the end it failed searching for that specific episode because Jackett was only returning the first 100 torrents, while animebytes has far more results for One Piece.

**Comment by ilike2burnthing** [comment]
I don't have an AnimeBytes account, but I don't see why it would be any different than any other indexer. I can't repeat that behaviour with RARBG or Nyaa, with a standard series or anime, searching automatically or manually, searching a series, season, or episode, or even specifically One Piece S21E146 (1037) as above.

The OP in the linked issue states this is seemingly random, even down to specific episodes in a series, so I could waste my day trying and still fail apparently.

The response you linked to isn't a Sonarr dev, they have made 2 commits, one of which was just plain text. They also just appear to be explaining what the user is seeing based on the logs, not necessarily that it should be happening.

The two devs that do reply don't actually explain why offset is being used, nor why it's seemingly random in its application. I suggest asking on their alternate support channels mentioned (ideally their forum or subreddit, in case you need to link to the answer).

**Comment by bakerboy448** [comment]
> With that said, Sonarr wouldn't use it even if it were available, it wants a maximum of 100 results per indexer, and just makes one request per keyword.

Incorrect.  Sonarr will paginate up to 10 pages (1000 results).  This is based on the number of results indicated in the response as per spec.   There needs to be more than 100 results returned. [Test Ref](https://github.com/Sonarr/Sonarr/blob/6660e0c3f37fade91d39141b2830dba4e20a46e2/src/NzbDrone.Core.Test/IndexerTests/SeasonSearchFixture.cs#L64)

Newznab/Torznab spec is pagination, are you saying that jackett fails to implement torznab spec?

>   The offset is pagination. It grabs 0-100 results. Then it grabs 101-200. And so on. 1000 results total, 10 API hits.

_Originally posted by @PearsonFlyer in https://github.com/Sonarr/Sonarr/issues/5136#issuecomment-1229183254_
      
is accurate.

If jackett's trackers / implementation does not support torznab offset, then their caps should be updated to indicate as such.
<https://github.com/Sonarr/Sonarr/blob/6660e0c3f37fade91d39141b2830dba4e20a46e2/src/NzbDrone.Core.Test/IndexerTests/SeasonSearchFixture.cs#L74>

It appears setting caps.PageSize to 0 would disable paging

**Comment by ilike2burnthing** [comment]
As far as I can see `PageSize` has never been implemented as a cap and exists nowhere in our code. The only uses of `LimitsDefault` and `LimitsMax` are for BTN, BHD, and the MetaIndexer.

Relevant open issues - https://github.com/Jackett/Jackett/issues/250 and https://github.com/Jackett/Jackett/issues/975.

**Comment by bakerboy448** [comment]
so #250 is the root cause of this specific issue given \*Arr needs to be told to not paginate.  This can be bandaided with a default value of 0 I guess.

#975 would need to be implemented in parallel with a proper #250 fix. to fully resolve this then.
